src/training/scheduler.cpp

```diff
@@ -0,0 +1,32 @@
+#include "scheduler.h"
+#include <signal.h>
+#include <cassert>
+
+namespace marian {
+bool Scheduler::sigterm_{false};
+
+void
+Scheduler::
+signalHandler_(int sig) {
+  // Note: sys_siglist[sig] or stdsignal() describe the effect (e.g.,
+  // 'Terminated' rather than provide the signal name (which are #define(s)
+  // in signal.h), so we have to do custom log messages here.
+  switch (sig) {
+  case SIGTERM: // save models and exit
+    LOG(info, "[training] Scheduler received signal SIGTERM");
+    Scheduler::sigterm_ = true;
+    break;
+  default:
+    ABORT("No action defined for signal {}", sig);
+  }
+}
+
+
+void
+Scheduler::
+installSignalHandlers_() {
+  // TODO: use sigaction instead of signal
+  signal(SIGTERM, Scheduler::signalHandler_);
```

Comment:
Why `sigaction` would be better than `signal` and what is the reason it's only a TODO? A quick googling suggests that `sigaction` is not supported on Windows?

Reply:
> Why sigaction would be better than signal
https://stackoverflow.com/questions/231912/what-is-the-difference-between-sigaction-and-signal

> what is the reason it's only a TODO
Convenience of implementation. Old dogs, new tricks and so forth.

Reply:
OK, would you mind adding this link into that TODO for futher reference? Thanks